test/common/schemaform/save-load-actions.unit.spec.js

```diff
@@ -0,0 +1,342 @@
+      const status = LOAD_STATUSES.success;
+      const action = setFetchFormStatus(status);
+
+      expect(action.type).to.equal(SET_FETCH_FORM_STATUS);
+      expect(action.status).to.equal(status);
+    });
+  });
+  describe('setInProgressForm', () => {
+    it('should return action', () => {
+      const data = {};
+      const action = setInProgressForm(data);
+
+      expect(action.type).to.equal(SET_IN_PROGRESS_FORM);
+      expect(action.data).to.equal(data);
+    });
+  });
+  describe('loadInProgressDataIntoForm', () => {
+    it('should return action', () => {
+      const action = loadInProgressDataIntoForm();
+
+      expect(action.type).to.equal(LOAD_DATA_INTO_FORM);
+    });
+  });
+  describe('migrateFormData', () => {
+    it('should return migrated data', () => {
+      const data = {
+        field: 'stuff'
+      };
+      const migrations = [
+        (savedData) => {
+          savedData.field = savedData.field.toUpperCase(); // eslint-disable-line no-param-reassign
+          return savedData;
+        }
+      ];
+      const migratedData = migrateFormData(data, 0, migrations);
+
+      expect(migratedData).to.eql({
+        field: 'STUFF'
+      });
+    });
+    it('should migrate multiple times', () => {
+      const data = {
+        field: 'stuff'
+      };
+      const migrations = [
+        (savedData) => {
+          savedData.field = savedData.field.toUpperCase(); // eslint-disable-line no-param-reassign
+          return savedData;
+        },
+        (savedData) => {
+          savedData.field = `${savedData.field} to do`; // eslint-disable-line no-param-reassign
+          return savedData;
+        }
+      ];
+      const migratedData = migrateFormData(data, 0, migrations);
+
+      expect(migratedData).to.eql({
+        field: 'STUFF to do'
+      });
+    });
+  });
+  describe('saveInProgressForm', () => {
+    beforeEach(setup);
+    afterEach(teardown);
+
+    it('dispatches a no-auth if the user has no session token', (done) => {
+      const thunk = saveInProgressForm('hca', 0, 'some/path', {});
+      const dispatch = sinon.spy();
+      delete sessionStorage.userToken;
+
+      Promise.all([thunk(dispatch)]).catch(() => {
+        expect(dispatch.calledWith(setSaveFormStatus(SAVE_STATUSES.noAuth))).to.be.true;
+        expect(dispatch.calledWith(setSaveFormStatus(SAVE_STATUSES.pending))).to.be.false;
+        done();
+      });
+    });
+    it('dispatches a pending', (done) => {
+      const thunk = saveInProgressForm('hca', 0, 'some/path', {});
+      const dispatch = sinon.spy();
+
+      Promise.all([thunk(dispatch)]).then(() => {
+        expect(dispatch.calledWith(setSaveFormStatus(SAVE_STATUSES.pending))).to.be.true;
+        done();
+      });
+    });
+    it('calls the api to save the form', (done) => {
+      const thunk = saveInProgressForm('hca', 0, 'some/path', {});
+      const dispatch = sinon.spy();
+
+      Promise.all([thunk(dispatch)]).then(() => {
```

Comment:
Y'know, I think that was just me being dumb. `thunk(dispatch).then()` / `.catch()` should be fine. Originally, I had to have a way to `.then()` or `.catch()` when I wasn't returning a Promise, but that didn't work out so well after all anyhow, so :man_shrugging: